### PR TITLE
Use Python 3.9 for Mercurial, related ports and their dependents

### PR DESCRIPTION
**Pull request by danchr**
#### Description

This switches the Mercurial port to using Python 3.9. Although Mercurial and most of its dependents don't affect all _that_ many ports, this does not apply to TortoiseHg, which uses PyQt and an awful lot of other ports. Hence this PR.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Review by michaelld — APPROVED** [review]
LGTM for qscintilla, py-pyqt4, py-sip. Given the complexity of this PR, I might end up adding on Py39 support to these before this PR is merged ... we'll see!

**Comment by ryandesign** [comment]
I'm not sure that we should switch any ports to using python39 exclusively or by default at this point. Yes we should add python39 support to ports that can support multiple python versions, but for ports that use a single version of python we should use the version that the python portgroup defaults to, which at this time, since python39 was just introduced, [is still python38](https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/python-1.0.tcl#L76).

Since python39 was just added, most of the python module ports don't support python39 yet. This means that many ports that use python38 cannot be simply switched to python39. We don't want the situation where some ports require python39 and others require python38; users have in the past complained about such perceived superfluous dependencies.

**Comment by danchr** [comment]
> Since python39 was just added, most of the python module ports don't support python39 yet. This means that many ports that use python38 cannot be simply switched to python39. We don't want the situation where some ports require python39 and others require python38; users have in the past complained about such perceived superfluous dependencies.

Ah, that makes sense; I can just apply all the subport additions, and hold off on actually switching over?

**Pull request merged**
